source/Halibut/Diagnostics/ExceptionReturnedByHalibutProxyExtensionMethod.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.IO;
+using System.Net.Sockets;
+using Halibut.Exceptions;
+using Halibut.Transport;
+using Halibut.Transport.Protocol;
+using Halibut.Transport.Proxy.Exceptions;
+
+namespace Halibut.Diagnostics
+{
+
+    public static class ExceptionReturnedByHalibutProxyExtensionMethod
+    {
+        public static HalibutNetworkExceptionType IsNetworkError(this Exception exception)
+        {
+            if (exception is NoMatchingServiceOrMethodHalibutClientException)
+            {
+                return HalibutNetworkExceptionType.NotANetworkError;
+            }
+
+            if (IsErrorInService(exception))
+            {
+                return HalibutNetworkExceptionType.NotANetworkError;
+            }
+
+            if (exception is UnexpectedCertificateException
+                || exception is FileNotFoundException)
```

Comment:
This FileNotFoundException is a bit of a mess, because the DataStream itself has a "Action" that writes to the given stream (ie the network) we don't know if the error in a DataStream because writing to the given stream (ie the network) caused an error OR if something else caused an error in a DataStream.

At least right now a `FileNotFoundException` is always going to be some sort of non network error so this works for the errors we have seen in server.